### PR TITLE
[Merged by Bors] - chore: remove more vars

### PR DESCRIPTION
**Pull request by adomani**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/List/Basic.lean [hunk 1]
```diff
@@ -2278,8 +2276,7 @@ theorem disjoint_reverse_right {l₁ l₂ : List α} : Disjoint l₁ l₂.revers
 end Disjoint
 
 section lookup
-
-variable {α β : Type*} [BEq α] [LawfulBEq α]
+variable [BEq α] [LawfulBEq α]
```
Comment (b-mehta) [inline Mathlib/Data/List/Basic.lean:2279]: The removal of alpha here is surprising to me. Is this change secretly changing the implicitness of alpha below? I don't really see how else the old and new version could both compile

Comment (adomani) [reply Mathlib/Data/List/Basic.lean:2279]: They were already defined and already implicit, so this was redundant here.

Comment (b-mehta) [reply Mathlib/Data/List/Basic.lean:2279]: Huh, I thought that `variables` which were defined and not changing explicitness gave an error. Thanks for explaining!

**Comment by b-mehta** [comment]
bors d+

File: Mathlib/Data/List/Basic.lean (hunk 1 above)
Comment (adomani) [reply Mathlib/Data/List/Basic.lean:2279]: To be completely explicit: a *binder update* gives an error, but *shadowing* does not:
```
variable {α : Type}
variable {α : Type} -- no problem!
variable {α} -- redundant binder annotation update
```
This command was not a binder update, it was shadowing variables defined at the beginning of the file.

Comment (b-mehta) [reply Mathlib/Data/List/Basic.lean:2279]: Ah, that is indeed exactly the confusion I had. Thanks again!

**Pull request closed without merging**